lightning/src/ln/channel.rs

```diff
@@ -695,6 +695,13 @@ pub(super) struct Channel<Signer: Sign> {
 
 	/// This channel's type, as negotiated during channel open
 	channel_type: ChannelTypeFeatures,
+
+	// Our counerparty can offer us SCID aliases which they will map to this channel when routing
+	// outbound payments. These can be used in invoice route hints to provide privacy of which
+	// on-chain transaction is ours.
+	// We only bother storing the most recent SCID alias at any time, though our counterparty has
+	// to store all of them.
+	latest_inbound_scid_alias: Option<u64>,
```

Comment:
Hmm, that isn't the only place we use them post-0conf - they're general aliases, and someone could totally use them to route over a channel prior to confirmation.

Reply:
I thought about this a bit more and I still kinda like `inbound` as the terminology here - it matches all the ways I'd think about it (its a value we received from our counterparty, and its also the value used for payments routed inbound towards us). I still kinda want to avoid terming it "route hint" cause users may use it outside of explicit invoice contexts (eg by handing someone a last-hop explicitly via some other protocol) and "route hint" sounds very invoice-y, even though the alternative uses I can see immediately are all similar concepts.

I guess I'm not quite sure why the inbound/outbound split is confusing - it seems to line up with all the things I can think of.

Reply:
Sg, I guess not that it's confusing per se but I thought the rename would make it more obvious that the correct alias is being used in a given context. Fine to let it rest